java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture2/tasks/_2/Number2.java

```diff
@@ -0,0 +1,164 @@
+        System.out.println("Set direction where your robot is going. Default direction is SOUTH");
+        System.out.println("0. SOUTH,\n1. NORTH,\n2. WEST,\n3. EAST");
+        Direction direction = Direction.values()[Integer.parseInt(scanner.next())];
+        System.out.println(direction.toString() + "\n");
+
+        System.out.println("1.Move Forward.\n2.Change Direction;\n3.Get coordinates;\n4.Get directions;\n5.Exit");
+
+        while (!exit) {
+            switch (scanner.nextInt()) {
+                case 1 -> {
+                    System.out.println("Type N steps");
+                    int steps = scanner.nextInt();
+                    robotGame.moveForward(steps);
+                }
+                case 2 -> {
+                    System.out.println("1.Turn Left;\n2.Turn Right;");
+                    if (scanner.nextInt() == 1) {
+                        robotGame.turnLeft();
+                        System.out.println("Turned left");
+
+                    } else {
+                        robotGame.turnRight();
+                        System.out.println("Turned right");
+                    }
+                }
+                case 3 -> System.out.println(robotGame.getCoordinates());
+                case 4 -> robotGame.getDirection();
+                case 5 -> {
+                    System.out.println("Shutting down");
+                    exit = true;
+                }
+            }
+        }
+    }
+}
+
+enum Direction {
+    SOUTH,
+    NORTH,
+    WEST,
+    EAST
+}
+
+class RobotGame {
+
+    private final int width;
+    private final int height;
+    private int x;
+    private int y;
+    Direction direction;
+
+    public RobotGame(int width, int height, int x, int y, Direction direction) {
+        this.width = width;
+        this.height = height;
+        this.x = x;
+        this.y = y;
+        this.direction = direction;
+    }
+
+    public String getRoomSize() {
+        return "Width: " + width + " " + "Height: " + height;
+    }
+
+    public String getCoordinates() {
+        return "X: " + x + " " + "Y: " + y;
+    }
+
+    public void getDirection() {
+        System.out.println(direction);
+    }
+
+    public void moveForward(int n) {
+        if (n > height || n > width) {
+            System.out.println("Amount of steps are greater than a width or height of a room");
+        }
+        else if ((n + y) > height || (n + x) > width) {
+            System.out.println("Can't move forward. Out of bounds.");
+        }
+        else {
+            if (direction == Direction.SOUTH) {
+                if (y < height) {
+                    y += n;
+                }
+                if (y == height) {
+                    System.out.println("WARNING. OUT OF RANGE");
+                }
+            } else if (direction == Direction.NORTH) {
+                if (y > 0) {
+                    y -= n;
+                }
+                if (y < n) {
```

Comment:
my comment 
```
if y < n, we'll move out of the bounds
```
was an explanation that it's not enough to check for `y == 0`, you need to check if `y < 0` (0 is the valid value).

Please remove the condition you've added (y < n) and change the (y == 0) condition. Same should be done for all directions.